virttest/utils_disk.py

```diff
@@ -372,15 +373,22 @@ def create_partition_linux(session, did, size, start,
     :param start: partition beginning at start. e.g. 0M
     :param part_type: partition type, primary extended logical
     :param timeout: Timeout for cmd execution in seconds.
+    :return: the created partition.
     """
     size = utils_numeric.normalize_data_size(size, order_magnitude="M") + "M"
     start = utils_numeric.normalize_data_size(start, order_magnitude="M") + "M"
     end = str(float(start[:-1]) + float(size[:-1])) + size[-1]
     partprobe_cmd = "partprobe /dev/%s" % did
-    mkpart_cmd = 'parted -s "%s" mkpart %s %s %s'
-    mkpart_cmd %= ("/dev/%s" % did, part_type, start, end)
-    session.cmd(mkpart_cmd)
+    mkpart_cmd = 'parted -s "/dev/%s" mkpart %s %s %s print' % (did, part_type,
+                                                                start, end)
+    output = session.cmd(mkpart_cmd)
+    partition = did + re.findall(r'\s+(\d+)\s+\d+', output, re.M)[-1]
     session.cmd(partprobe_cmd, timeout=timeout)
+    if not wait.wait_for(
+            lambda: not session.cmd_status_output('ls /dev/%s' % partition)[0],
+            step=0.5, timeout=30):
+        raise exceptions.TestError('Failed to create %s.' % partition)
+    return partition
```

Comment:
for \"create_partition_windows\", no return value, I think it should keep the same rule, in order to provides consistency in \"create_partition\"

Reply:
I believe there is no need to return the partition label.

Reply:
I believe that returning the partition label is better, since the places where call this function will not get it again. like the configure_empty_linux_disk function doesn't need to get_linux_disks twice.

Reply:
If so, please elaborate this change in your commit message.